File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/CommonDialogs/OpenFileDialog/Programming/FilePage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.CommonDialogs.OpenFileDialog.Programming
{
    public partial class FilePage : UserControl
    {
        public FilePage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles the click event of the open file dialog box button.
        /// Opens the demonstrated OpenFileDialog dialog box.
        /// </summary>
        private void mobjOpenFileDialogButton_Click(object sender, EventArgs e)
        {
            this.mobjDemoOpenFileDialog.ShowDialog();
        }

        /// <summary>
        /// Handles FileOK event of the demonstrated OpenFileDialog dialog box.
        /// </summary>
        private void mobjDemoOpenFileDialog_FileOk(object sender, CancelEventArgs e)
        {
            // Change text of TextBoxes to the file name and the file size.
            this.mobjOrigFileNameTextBox.Text = ((Gizmox.WebGUI.Common.Resources.HttpPostedFileHandle)(this.mobjDemoOpenFileDialog).File).OriginalFileName;
            this.mobjFileNameTextBox.Text = this.mobjDemoOpenFileDialog.File.FileName;
            this.mobjFileSizeTextBox.Text = this.mobjDemoOpenFileDialog.File.ContentLength.ToString();
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/CommonDialogs/OpenFileDialog/Programming/FilePage.vb
Namespace CompanionKit.Controls.CommonDialogs.OpenFileDialog.Programming

	Public Class FilePage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

		End Sub

        ''' <summary>
        ''' Handles the click event of the open file dialog box button.
        ''' Opens the demonstrated OpenFileDialog dialog box.
        ''' </summary>
        Private Sub mobjOpenFileDialogButton_Click(sender As Object, e As EventArgs) Handles mobjOpenFileDialogButton.Click
            Me.mobjDemoOpenFileDialog.ShowDialog()
        End Sub

        ''' <summary>
        ''' Handles FileOK event of the demonstrated OpenFileDialog dialog box.
        ''' </summary>
        Private Sub mobjDemoOpenFileDialog_FileOk(sender As Object, e As ComponentModel.CancelEventArgs) Handles mobjDemoOpenFileDialog.FileOk
            ' Change text of TextBoxes to the file name and the file size.
            Me.mobjOrigFileNameTextBox.Text = DirectCast((Me.mobjDemoOpenFileDialog).File, Gizmox.WebGUI.Common.Resources.HttpPostedFileHandle).OriginalFileName
            Me.mobjFileNameTextBox.Text = Me.mobjDemoOpenFileDialog.File.FileName
            Me.mobjFileSizeTextBox.Text = Me.mobjDemoOpenFileDialog.File.ContentLength.ToString()
        End Sub
    End Class

End Namespace
